### PR TITLE
feat/AB#76296_remove-duplication-of-page-step-actions

### PR DESCRIPTION
**Pull request by estelafs**
# Description
Created PageSettingsComponent and old icon-modal is now PageIconComponent (no longer opened on a modal)

Added a tooltip next to the modal title to inform that changes are saved automatically, without there being a save button

PageSettingsComponent has shared page/actions steps: edit icon, page visibility, duplicate page and edit access (respecting permission conditions and checks if action is available only for pages, like the visibility and duplicate)

Using the workflow service and the application service for all shared mutations. Deleted unused mutations and functions. Moved the method responsible for the mutation notifications to the application service, where multiple other old methods also benefit from it

Updated EditAccessComponent to to work whether opened in a modal or not (resources still need the access edition on a modal)

## Useful links

- Please insert link to ticket: https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/76296

## Type of change
- [X] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?
On different application pages (dashboard, form and workflow) and on different step types (form and dashboard), open the settings modal by clicking on the gear icon next to the title and the available actions and the saving is done automatically on every change

## Screenshots
image

Dashboard:
[dashboard.webm](https://github.com/ReliefApplications/oort-frontend/assets/28535394/4a410464-d293-4719-8282-988efc3d704e)

Form:
[form.webm](https://github.com/ReliefApplications/oort-frontend/assets/28535394/4ec66d9a-90f0-4c8f-9a01-6820ed5bbc80)

Workflow + dashboard step + form step:
[workflow.webm](https://github.com/ReliefApplications/oort-frontend/assets/28535394/faa73bd8-5364-455a-a63c-710390d7b1ba)

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by MwanPygmay — APPROVED** [review]
merged, tested, quickly read code, seems good to me

File: libs/shared/src/i18n/en.json [hunk 1]
```diff
@@ -538,6 +538,10 @@
 					"plural": "The pages were successfully reordered.",
 					"singular": "The page was successfully reordered."
 				},
+				"settings": {
+					"selectIcon": "Select an icon",
+					"settingsTooltip": "Changes are automatically saved"
```
Comment (AntoineRelief) [inline libs/shared/src/i18n/en.json]: @estelafs 
cc @MwanPygmay 
tooltips keys need to be stored under a "tooltip" property
some other examples are in the translation files

File: apps/back-office/src/app/components/page-icon/page-icon.component.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { IconPickerModule } from '@oort-front/shared';
+import { TranslateModule } from '@ngx-translate/core';
+import { FormControl, ReactiveFormsModule } from '@angular/forms';
+
+/**
+ * Icon chooser of pages / steps.
+ * Use shared icon picker.
+ */
+@Component({
+  selector: 'app-page-icon',
```
Comment (AntoineRelief) [inline apps/back-office/src/app/components/page-icon/page-icon.component.ts]: @estelafs 
cc @MwanPygmay 
the component is called app-page-icon but is also used for steps
the previous name was better

**Comment by AntoineRelief** [comment]
@estelafs 
cc @MwanPygmay 
merged, you can see the changes 👍 

the code is fine, and the features working
I just want to make the code the cleaner as possible

**Pull request merged**